Course Recommendation Frontend

andyjiang3 opened the pull request:
• Added course rec frontend

Currently using search result for course recommendations
Need to link with the backend when it is finished


<img width="374" alt="Screen Shot 2021-03-28 at 1 05 13 PM" src="https://user-images.githubusercontent.com/44344658/112761011-147e0880-8fc7-11eb-9d48-e0d23ef50c3a.png">
<img width="377" alt="Screen Shot 2021-03-28 at 1 04 55 PM" src="https://user-images.githubusercontent.com/44344658/112761012-1778f900-8fc7-11eb-9f3f-9059419a6750.png">

dlike230 commented on frontend/plan/components/selector/Selector.tsx:
```diff
@@ -58,20 +66,79 @@ const Selector: FunctionComponent<SelectorProps> = ({
+    //delete func - does nothing for now
+    const onClickDelete = () => {};
+
+    useEffect(() => {
```
This useEffect is missing some deps (such as setRecCourses)

dlike230 commented on frontend/plan/components/selector/Selector.tsx:
```diff
@@ -58,20 +66,79 @@ const Selector: FunctionComponent<SelectorProps> = ({
+        };
+
+        //@ts-ignore Not sure why this is not typechecking
+        fetch(`/api/plan/recommendations/`, requestOptions).then((res) =>
```
We have a helper function for this, I believe it's called doApiRequest

dlike230 reviewed (changes requested):
Overall, great work! Only thing (other than my inline comments) is I'm not sure how we should go about updating the recommendations displayed (if at all) when a user's schedules change. Right now, if a user adds new classes that would affect their recommendations, that would have no effect on the recommendations displayed until the user refreshed the page. We could update recommndations every time the schedule is updated, but that might result in too many API calls. I'm tagging @mureytasroc and @davish to get their input.

mureytasroc commented:
> Overall, great work! Only thing (other than my inline comments) is I'm not sure how we should go about updating the recommendations displayed (if at all) when a user's schedules change. Right now, if a user adds new classes that would affect their recommendations, that would have no effect on the recommendations displayed until the user refreshed the page. We could update recommndations every time the schedule is updated, but that might result in too many API calls. I'm tagging @mureytasroc and @davish to get their input.
> 
> **Edit:** We should probably also remove the "x" buttons next to recommendations, since they don't do anything.

What do you think about adding a refresh button to "refresh recommendations"?

dlike230 commented:
> > Overall, great work! Only thing (other than my inline comments) is I'm not sure how we should go about updating the recommendations displayed (if at all) when a user's schedules change. Right now, if a user adds new classes that would affect their recommendations, that would have no effect on the recommendations displayed until the user refreshed the page. We could update recommndations every time the schedule is updated, but that might result in too many API calls. I'm tagging @mureytasroc and @davish to get their input.
> > **Edit:** We should probably also remove the "x" buttons next to recommendations, since they don't do anything.
> 
> What do you think about adding a refresh button to "refresh recommendations"?

Seems like the best solution. Maybe for now we can just have recs update at a scheduled interval if changes have been made to the schedule?

andyjiang3 commented:
• Added star next to rec courses in the search list
• Added error messages
• Added refresh button 

<img width="374" alt="Screen Shot 2021-04-01 at 7 53 38 PM" src="https://user-images.githubusercontent.com/44344658/113365606-bd5c9880-9324-11eb-8325-62093adabd47.png">

dlike230 commented on frontend/plan/components/recomendations/RecBanner.tsx:
```diff
@@ -15,35 +18,65 @@ const BannerContainer = styled.div<{ collapse: boolean }>`
 
-const RecBanner = ({ show, setShow }: RecBannerProps) => {
+const RecBanner = ({ show, setShow, setRefresh }: RecBannerProps) => {
+    const [disabled, setDisabled] = useState(false);
```
For the sake of clarity, let's call this `refreshDisabled`

dlike230 commented on frontend/plan/components/recomendations/RecInfo.tsx:
```diff
@@ -45,17 +38,30 @@ const RecInfo = () => {
+                    recommendations by users of Penn Mobile). Refreshing the
+                    page or pressing the refresh button above the
+                    recommendations section will take into account any changes
+                    you have made to your PCP schedules since the last refresh.
```
Replace PCP with Penn Course Plan

dlike230 reviewed (changes requested):
Just a couple really small things

dlike230 reviewed (approved):
LGTM!
